Automattic/wp-calypso
# Submit already WPCOM survey to endpoint, route user to survey step when WPCOM detected

Related to https://github.com/Automattic/wp-calypso/issues/95021

## Proposed Changes

* Brings users to the survey step if WPCOM is detected
* Submits the data to the backend after taking survey
* Disable form when submitting data

## Why are these changes being made?


* To understand why the user submitted a site which is already on WPCOM

## Testing Instructions



Note: **_Continue Anyways_** path means if you submit a site with wrong URL, wrong creds (not invalid, only wrong) or a site that's already on WPCOM, we show error messages, but the submit button changes to **_Continue Anyways_** in the Credentials form. Which lets you submit the ticket despite having those errors.

- Apply this patch in your sandbox - 
- Make sure WPCOM requests are pointed to your sandbox
- Proceed through the migration flow.
- On the "How do you want to migrate?" step, choose "Do it for me".
- In the credentials form, test the following scenarios -

# Enter a non WPCOM WordPress site
- Try with wrong credentials and you should be able to submit in the Continue Anyways path.
- Try with the right credentials and the submission should happen without any issue
- You should not see the survey form
- Take the survey by selecting one or more fields, try also adding the Other details and submit
- You should be taken to the next page
- On ZD, you should be able to see a comment with the result of your survey

# Enter a WPCOM site
- Try with right/wrong credentials and you should be able to submit in the Continue Anyways path, you should see an error about the site being on WPCOM already
- If you submit using Continue anyways, it shouldn't farther show you any error even you provide wrong creds
- You should see the survey form

# Enter a non WP site
- You should see an error under the URL field
- You should be able to submit in the Continue anyways path
- You should not see the survey form

https://github.com/user-attachments/assets/ab95f85f-7e44-47c6-ba81-b72c21f3717e

## Pre-merge Checklist
